crates/libs/result/src/bstr.rs

```diff
@@ -1,36 +1,26 @@
 use super::*;
+use core::ops::Deref;
 
 #[repr(transparent)]
 pub struct BasicString(*const u16);
 
-impl BasicString {
-    pub fn is_empty(&self) -> bool {
-        self.len() == 0
-    }
+impl Deref for BasicString {
+    type Target = [u16];
 
-    pub fn len(&self) -> usize {
-        if self.0.is_null() {
+    fn deref(&self) -> &[u16] {
```

Comment:
What's the reason for this signature rather than its more conventional spelling

```rust
fn deref(&self) -> &Self::Target
```

?

I'm just curious if there's a particular reason.

Reply:
No reason, just seemed more succinct in this case.